pkg/controllers/gateway/gatewayclass_controller.go

```diff
@@ -18,14 +18,27 @@ package gateway
 
 import (
 	"context"
+	"fmt"
+	"strings"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/slice"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/log"
-	gateway "sigs.k8s.io/gateway-api/apis/v1beta1"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
 )
 
+const (
+	ControllerName = "kuadrant.io/mctc-gw-controller"
+)
+
+func getSupportedClasses() []string {
```

Comment:
probably want to consider externalising these in the future

Reply:
Agreed.
The pattern of the controller creating & managing it's own classes would hit on this.